Sources/StreamChatUI/ChatMessageList/ChatMessageListVC.swift

```diff
@@ -654,3 +712,123 @@ open class ChatMessageListVC: _ViewController,
         !(touch.view is UIControl)
     }
 }
+
+// MARK: - Handle Message Updates
+
+private extension ChatMessageListVC {
```

Comment:
The logic here is pretty much the same as we already had (Especially with the most up-to-date implementation). I just extracted it from the `ChatMessageListView` to the `ChatMessageListVC` so that it can be easily testable and accommodate for this new feature. I've also broken it down into multiple functions to that it is easier to read. 

I've also added test coverage for all this logic 🙂 Which was previously untested, and we could easily break things.